tcpserver.c

```diff
@@ -163,3 +165,44 @@ void close_server(void)
     int status = close(own_socket);
     printf("Attempted closing server with status: %d\n", status);
 }
+
+/**
+* Add a new element to client_sockets
+* @param new_element element to be appended to client_sockets
+*/
+void add_to_array(int new_element) 
```

Comment:
I suggest that you remove the realloc part and uses the calloc as suggested earlier because it only saves space for the number of clients supported. Also, calloc allocated memory as needed.